admin-dev/themes/new-theme/js/pages/improve/design_positions/positions-list-handler.js

```diff
@@ -0,0 +1,219 @@
+        self.$panelSelectionMultipleSelection.hide();
+      } else {
+        self.$panelSelection.show();
+        self.$panelSelectionSingleSelection.hide();
+        self.$panelSelectionMultipleSelection.show();
+        $('#modules-position-selection-count').html($checkedCount);
+      }
+    });
+
+    self.$panelSelection.find('button').click(() => {
+      $('button[name="unhookform"]').trigger('click');
+    });
+
+    self.$hooksList = [];
+    $('section.hook-panel .hook-name').each(function () {
+      const $this = $(this);
+      self.$hooksList.push({
+        'title': $this.html(),
+        'element': $this,
+        'container': $this.parents('.hook-panel')
+      });
+    });
+
+    self.$showModules.select2();
+    self.$showModules.on('change', () => {
+      this.modulesPositionFilterHooks();
+    });
+
+    self.$hookPosition.on('change', () => {
+      this.modulesPositionFilterHooks();
+    });
+
+    self.$hookSearch.on('input', () => {
+      this.modulesPositionFilterHooks();
+    });
+
+    $('.hook-checker').on('click', function() {
+      $(`.hook${$(this).data('hook-id')}`).prop('checked', $(this).prop('checked'));
+    });
+  }
+
+  handleSortable() {
+    const self = this;
+
+    $('.sortable').sortable({
+      forcePlaceholderSize: true,
+      start: function(e, ui) {
+        $(this).data('previous-index', ui.item.index());
+      },
+      update:  function(e, ui) {
+        const $ids = ui.item.attr('id').split('_');
+        const $data = {
+          hookId: $ids[0],
+          moduleId: $ids[1],
+          way: ($(this).data('previous-index') < ui.item.index()) ? 1 : 0,
+          positions: [],
+        };
+
+        $.each(e.target.children, function(index, element) {
+          $data.positions.push($(element).attr('id'));
+        });
+
+        $.ajax({
+          type: 'POST',
+          headers: {'cache-control': 'no-cache'},
+          url: self.$modulePositionsForm.data('update-url'),
+          data: $data,
+          success: () => {
+            let start = 0;
+
+            $.each(e.target.children, function(index, element) {
+              $(element).find('.index-position').html(++start);
+            });
+
+            window.showSuccessMessage(window.update_success_msg);
+          }
+        });
+      },
+    });
+  }
+
+  modulesPositionFilterHooks() {
+    const self = this;
+    const $hookName = self.$hookSearch.val();
+    const $moduleId = self.$showModules.val();
+    const $position = self.$hookPosition.prop('checked');
+    const $regex = new RegExp(`(${$hookName})`, 'gi');
+
+    for (let $id = 0; $id < self.$hooksList.length; $id++) {
+      self.$hooksList[$id].container.toggle($hookName === '' && $moduleId === 'all');
+      self.$hooksList[$id].element.html(self.$hooksList[$id].title);
+      self.$hooksList[$id].container.find('.module-item').removeClass('highlight');
+    }
+
+    if ($hookName !== '' || $moduleId !== 'all') {
```

Comment:
I know we're not supposed to redo this part, but it's really difficult to understand every test/condition here For the most important ones, would to mind to extract meaningful functions instead?

Something like

```js
if (self.meaningfulCondition($hookname, $moduleId)) { ... }
```